src/intervals/construction.jl

```diff
@@ -200,10 +208,11 @@ to contain the number that was typed in.
 
 Otherwise it is the same as using the `Interval` constructor directly.
 """
-atomic(::Type{Interval{T}}, x) where {T} = interval(T, x)
-atomic(::Type{Interval{T}}, x::AbstractString) where {T} = parse(Interval{T}, x)
+atomic(::Type{Interval{T}}, x) where {T<:NumTypes} = interval(T, x)
+atomic(::Type{Interval{T}}, x::AbstractString) where {T<:NumTypes} = parse(Interval{T}, x)
 
-function atomic(::Type{Interval{T}}, x::AbstractFloat) where {T}
+atomic(::Type{Interval{T}}, x::AbstractFloat) where {T<:Rational} = Interval{T}(lo, hi)
+function atomic(::Type{Interval{T}}, x::AbstractFloat) where {T<:AbstractFloat}
```

Comment:
The following is not the tightest result, though it's a *valid* result
```julia
julia> IntervalArithmetic.atomic(Interval{Float64}, 0.1)
Interval(0.09999999999999999, 0.10000000000000002)
```
I think this should be documented.

The tightest result is obtained with `IntervalArithmetic.atomic(Interval{Float64}, "0.1")`